pkg/controller/container-runtime-config/helpers.go

```diff
@@ -954,3 +1007,65 @@ func generateSigstoreRegistriesdConfig(clusterScopePolicies map[string]signature
 	}
 	return data, nil
 }
+
+func addScopeMirrorsSigstoreRegistriesdConfig(registriesDockerConfig map[string]dockerConfig, scope string, reg *sysregistriesv2.Registry, sigstoreAttachment dockerConfig) error {
+	if reg == nil {
+		return nil
+	}
+	dummyDigest := "sha256:aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+	dummyTag := "aaa"
+	dummyPath := "/a"
+	dummyPrefix := "matched"
+	addedDummy := false
+	hasWildcard := false
+	originalScope := scope
+
+	if strings.HasPrefix(scope, "*.") {
+		scope = strings.Replace(scope, "*", dummyPrefix, 1)
+		hasWildcard = true
+	}
+
+	if !strings.Contains(scope, "/") {
+		scope += dummyPath
+		addedDummy = true
+	}
+
+	scopeRef, err := reference.Parse(scope)
+	if err != nil {
+		return fmt.Errorf("error parsing scope %s: %w", scope, err)
+	}
+
+	repo := reference.TrimNamed(scopeRef.(reference.Named))
+
+	d, _ := digest.Parse(dummyDigest)
+	digestRef, err := reference.WithDigest(repo, d)
+	if err != nil {
+		return fmt.Errorf("error parsing digest name for scope %s: %w", scope, err)
+	}
+	tagRef, err := reference.WithTag(repo, dummyTag)
+	if err != nil {
+		return fmt.Errorf("error parsing tag name for scope %s: %w", scope, err)
+	}
+
+	digestSources, err := reg.PullSourcesFromReference(digestRef)
+	if err != nil {
+		return fmt.Errorf("error getting digest sources for scope %s: %w", scope, err)
+	}
+	tagSources, err := reg.PullSourcesFromReference(tagRef)
+	if err != nil {
+		return fmt.Errorf("error getting tag sources for scope %s: %w", scope, err)
+	}
+	for _, s := range append(digestSources, tagSources...) {
+		endpoint := s.Reference.Name()
+		if addedDummy {
+			endpoint = strings.TrimSuffix(endpoint, dummyPath)
+		}
+		if hasWildcard {
+			if endpoint == strings.Replace(originalScope, "*", dummyPrefix, 1) {
+				continue
+			}
+		}
```

Comment:
@mtrmac Could you review this?  We also need to add a check for `dummyPrefix`.

Reply:
(I’m afraid that my suggestion to use the c/image matching code has not been simpler — certainly not much simpler.)

---

To be explicit, the situation is a config entry with
- `prefix: *.example.com`
- `location` not set — OCP never does that for wild-carded scopes
- and perhaps some mirrors, perhaps not, e.g. this entry might exist because there was an “insecure” bool set

And in that case the source reference is returned as is.

Here, unlike the added `dummyPath`, I’m worried about collisions with user-defined names (e.g. a mirroring setup from `*.example.com` to `matched.example.com`). I don’t think just a string comparison can resolve that.

It seems to me that this needs a condition  `&& s.Endpoint.Location == ""` to rule out this case: mirror endpoints must, by definition, have a mirror `location`. So if we have `Location` set, the host name was user-supplied; if `Location` is unset, the host name comes from `dummyPrefix`.

(Alternatively, I think the `Location == ""` check could be the _only_ condition here, but then all primary endpoints would be ruled out and would have to be added at the top level. That would actually match the `addScopeMirrorsSigstoreRegistriesdConfig` function name a bit better — it would be adding _only_ mirrors, and that would be good enough for currently OCP-generated configs, OTOH it would be less correct for processing `registries.conf` _in general_, because top-level redirection (`prefix: "logical"; location: "actualServer") would not be handled. I think we can handle top-level redirection here “for free”, so it’s better to do so than to risk future surprises.)

Please make sure there’s a unit test for this situation, and document what the situation and what the condition is checking for. (Something vaguely like “if we had to add a `dummyPrefix` to process a wildcard `Registry` entry, we only want to configure the mirror endpoints, if any; not to add a new entry for exactly `dummyPrefix`+`originalScope`”). 

Also, this code would now _depend_ on the caller unconditionally configuring attachments for `registriesDockerConfig[policyScope]` — that also seems worth having a comment.